nscale.js

```diff
@@ -808,16 +811,73 @@ function startServer(args) {
   function start() {
     var logDir = nscaleRoot + '/log';
 
-    async.eachSeries(servers, function(server, cb) {
+    function launch(server, cb) {
       var log = logDir + '/' + server.replace('nscale-', '') + '.log';
-      exec(server + ' -c ' + config + ' > ' + log + ' 2>&1 &',
-           { stdio: 'inherit' }, cb);
-    }, function(err) {
-      if (!err) {
-        console.log('done!');
+      var cmd = 'exec ' + server + ' -c ' + config + ' > ' + log + ' 2>&1';
+      var child = exec(cmd, { stdio: 'inherit' }, function(err, stdout, stderr) {
+        if (err) { cb(err); }
+      });
+      cb(null, child);
+    };
+
+    /*  
+      return a status object that looks like this:
+      {running: true, listening: true}
+    */
+    function checkServerStatus(server, cb) {
+      var result = {};
+      var pidFile = path.join(nscaleRoot, 'data', '.' + server);
+      
+      if (fs.existsSync(pidFile)) {
+        var pid = Number(fs.readFileSync(pidFile));
+        var port = JSON.parse(fs.readFileSync(config)).modules.protocol.specific.port || 3223;
```

Comment:
All this logic was starting multiple server (kernel, api, web), but now we have only left the kernel.
Either each server has its own port (preferred), or you remove all the multi-server handling.